diff --git a/cpp/BoostParts/boost/python/detail/wrap_python.hpp b/cpp/BoostParts/boost/python/detail/wrap_python.hpp
index eaef784..b82065b 100644
--- a/cpp/BoostParts/boost/python/detail/wrap_python.hpp
+++ b/cpp/BoostParts/boost/python/detail/wrap_python.hpp
@@ -81,17 +81,27 @@
 //
 // Some things we need in order to get Python.h to work with compilers other
 // than MSVC on Win32
+// Includes fixes from https://svn.boost.org/trac/boost/attachment/ticket/11125/0001-Add-cygwin64-support-to-wrap_python.hpp.patch
 //
 #if defined(_WIN32) || defined(__CYGWIN__)
 # if defined(__GNUC__) && defined(__CYGWIN__)
 
-#  define SIZEOF_LONG 4
+#  if defined(__LP64__)
+#   define SIZEOF_LONG 8
+#  else
+#   define SIZEOF_LONG 4
+#  endif
 
 #  if PY_MAJOR_VERSION < 2 || PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION <= 2
 
 typedef int pid_t;
 
-#   define WORD_BIT 32
+#   if defined(__LP64__) 
+#    define WORD_BIT 64 
+#   else 
+#    define WORD_BIT 32 
+#   endif 
+
 #   define hypot _hypot
 #   include <stdio.h>
 
